openeo/rest/job.py

```diff
@@ -79,24 +80,61 @@ def list_results(self, type=None):
         # GET /jobs/{job_id}/results
         pass
 
-    def download_results(self, target: Union[str, pathlib.Path]) -> pathlib.Path:
-        """ Download job results."""
+    def download_results(self, target: Union[str, pathlib.Path] = None) -> list:
+        """
+        Download job results to the target folder path.
```

Comment:
This feels like a confusing API (especially the interplay between `target` and how many files there are in the result)

maybe we should provide instead:
- `download_result` to download a single result: fails when target is a folder or there are multiple results
- `download_results` is more generic: target should always be a folder

Reply:
Now there are these two functions in the job class.